stm32_base/hal/usart_stm32/usart_device_stm32.c

```diff
@@ -699,6 +699,16 @@ void USART1_IRQHandler(void)
         xQueueSendFromISR(xUsart0Rx, &cChar, &xTaskWokenByPost);
     }
 
+    if (USART_GetITStatus(USART1, USART_IT_ORE_RX) != RESET) {
+        /* Handle Overrun error
+         * This bit is set by hardware when the word currently being received in the shift register is
+           ready to be transferred into the RDR register while RXNE=1. An interrupt is generated if
+           RXNEIE=1 in the USART_CR1 register. It is cleared by a software sequence (an read to the
+           USART_SR register followed by a read to the USART_DR register) */
```

Comment:
Bit 3 ORE: Overrun error
This bit is set by hardware when the word currently being received in the shift register is
ready to be transferred into the RDR register while RXNE=1. An interrupt is generated if
RXNEIE=1 in the USART_CR1 register. It is cleared by a software sequence (an read to the
USART_SR register followed by a read to the USART_DR register).
0: No Overrun error
From page 1000 of stm32f4 reference manual

1: Overrun error is detected
Note: When this bit is set, the RDR register content will not be lost but the shift register will be
overwritten. An interrupt is generated on ORE flag in case of Multi Buffer
communication if the EIE bit is set.

From my understanding what's happening is that the ORE flag is set when the shift register is getting populated and the data hasn't been read yet. it's like a warning flag. when this happens, we never clear the interrupt in the way the processor wants it to be, so the interrupt fires continuously until watchdog occurs.